### ISSUE TITLE
always_require_non_null_named_parameters is obsolete in null-safety

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#2236]
```dart
class Example {
  final int number;

  // LINT: always_require_non_null_named_parameters
  Example({required this.content});
}
```

I believe the lint should probably just turn itself off in null-safety mode?

**Comment by bwilkerson** [comment]
Probably so, yes.

But I'm confused. I thought the lint would only be triggered if there's an `assert` of the form `assert(content != null)` (and I confirmed that in an opted out library). And indeed, I don't see a lint in an opted-in library with the code above (modified so that `content` is replaced by `number`.

(We might want a different lint to catch this case that I found while testing the reported case:
```
class Example {
  final int? number;

  Example({this.number}) : assert(number != null);
}
```
It seems like it would be better to write
```
class Example {
  final int? number;

  Example({int number}) : this.number = number;
}
```
I just don't know whether that case will come up often enough to warrant a lint.)
